src/System.Reflection.MetadataLoadContext/src/FromCoreRt/System/Reflection/Runtime/BindingFlagSupport/MemberPolicies.cs

```diff
@@ -0,0 +1,240 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Reflection.TypeLoading;
+using RuntimeTypeInfo = System.Reflection.TypeLoading.RoType;
+
+namespace System.Reflection.Runtime.BindingFlagSupport
+{
+    //=================================================================================================================
+    // This class encapsulates the minimum set of arcane desktop CLR policies needed to implement the Get*(BindingFlags) apis.
+    //
+    // In particular, it encapsulates behaviors such as what exactly determines the "visibility" of a property and event, and
+    // what determines whether and how they are overridden.
+    //=================================================================================================================
+    internal abstract class MemberPolicies<M> where M : MemberInfo
+    {
+        //=================================================================================================================
+        // Subclasses for specific MemberInfo types must override these:
+        //=================================================================================================================
+
+        //
+        // Returns all of the directly declared members on the given TypeInfo.
+        //
+        public abstract IEnumerable<M> GetDeclaredMembers(TypeInfo typeInfo);
+
+        //
+        // Returns all of the directly declared members on the given TypeInfo whose name matches filter. If filter is null,
+        // returns all directly declared members.
+        //
+        public abstract IEnumerable<M> CoreGetDeclaredMembers(RuntimeTypeInfo type, NameFilter filter, RuntimeTypeInfo reflectedType);
+
+        //
+        // Policy to decide whether a member is considered "virtual", "virtual new" and what its member visibility is.
+        // (For "visibility", we reuse the MethodAttributes enum since Reflection lacks an element-agnostic enum for this.
+        //  Only the MemberAccessMask bits are set.)
+        //
+        public abstract void GetMemberAttributes(M member, out MethodAttributes visibility, out bool isStatic, out bool isVirtual, out bool isNewSlot);
+
+        //
+        // Policy to decide whether "derivedMember" is a virtual override of "baseMember." Used to implement MethodInfo.GetBaseDefinition(),
+        // parent chain traversal for discovering inherited custom attributes, and suppressing lookup results in the Type.Get*() api family.
+        // 
+        // Does not consider explicit overrides (methodimpls.) Does not consider "overrides" of interface methods.
+        //
+        public abstract bool ImplicitlyOverrides(M baseMember, M derivedMember);
+
+        //
```

Comment:
Ideally these would be `<summary>` as well. I think it's fine to just open an issue and perhaps someone will clean these up later with a big regex.